### PR TITLE
alignment of FT0 modules

### PR DESCRIPTION
**Pull request by AllaMaevskaya**

### CODE REVIEW

File: Detectors/FIT/FT0/base/include/FT0Base/Geometry.h [hunk 1]
```diff
@@ -37,19 +43,38 @@ class Geometry
...
   static constexpr float ZdetC = 82;                        // number of radiatiors on C side
   static constexpr float ChannelWidth = 13.02;              // channel width in ps
   static constexpr float ChannelWidthInverse = 0.076804916; // channel width in ps inverse
+  static constexpr o2::detectors::DetID::ID getDetID() { return o2::detectors::DetID::FT0; }
+  void setAsideModules();
+  void setCsideModules();
+  const char* getSymbolicName(int index) const
+  {
+    /// return symbolic name of sensor
+    LOG(INFO) << o2::base::GeometryManager::getSymbolicName(getDetID(), index);
```
Comment (shahor02) [inline Detectors/FIT/FT0/base/include/FT0Base/Geometry.h]: why this log is needed here?

File: Detectors/FIT/FT0/base/include/FT0Base/Geometry.h [hunk 2]
```diff
@@ -37,19 +43,38 @@ class Geometry
...
+  {
+    /// Get a pointer to the TGeoPNEntry of a chip identified by 'index'
+    /// Returns NULL in case of invalid index, missing TGeoManager or invalid symbolic name
+    return o2::base::GeometryManager::getPNEntry(getDetID(), index);
+  }
 
  private:
   TVector3 mMCP[52];
+  TVector3 mAngels[28];
+  static std::string sSensorName; ///< Sensor name
```
Comment (shahor02) [inline Detectors/FIT/FT0/base/include/FT0Base/Geometry.h]: I don't see where this `sSensorName` is used. Btw, I guess you meant `mAngles`, not `mAngels`

File: Detectors/FIT/FT0/simulation/src/Detector.cxx [hunk 3]
```diff
@@ -413,6 +344,58 @@ TGeoVolume* Detector::SetCablesSize(int mod)
...
+
+  LOG(INFO) << "Add FT0 alignable volumes";
+
+  if (!gGeoManager) {
+    LOG(FATAL) << "TGeoManager doesn't exist !";
+    return;
+  }
+
+  TString volPath = Form("/cave_1/barrel_1");
+  TString symName = Form("/cave_1/barrel_1");
```
Comment (shahor02) [inline Detectors/FIT/FT0/simulation/src/Detector.cxx]: actually, the symbolic names are supposed to be more meaningful than the path. You can, surely, use the symname the same as the path, but then what is to point of repeating the same code of the string building?

File: Detectors/FIT/FT0/macros/FT0Misaligner.C [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+  if (!ccdbHost.empty()) {
+    std::string path = objectPath.empty() ? o2::base::NameConf::getAlignmentPath(detFT0) : objectPath;
+    LOGP(INFO, "Storing alignment object on {}/{}", ccdbHost, path);
+    o2::ccdb::CcdbApi api;
+    map<string, string> metadata; // can be empty
+    api.init(ccdbHost.c_str());   // or http://localhost:8080 for a local installation
+    // store abitrary user object in strongly typed manner
+    //  api.storeAsTFileAny(&params, path, metadata, tmin, tmax);
+  }
```
Comment (shahor02) [inline Detectors/FIT/FT0/macros/FT0Misaligner.C:51]: I see that you define as alignables all 52 modules, but here create "ideal" matrices for the A, C parts global positions only. Are those modules degrees of freedom supposed to be used?

Comment (AllaMaevskaya) [reply Detectors/FIT/FT0/macros/FT0Misaligner.C:51]: I don't think is could be any freedom, may be within tens microns. I will ask our engineers. Detector is install and survey numbers for A side were published yesterday evening.  But they measured only global position for  whole FIT-A detector. I prefer to create only matrix for whole A and C sides and keep modules at their "ideal" position.

Comment (shahor02) [reply Detectors/FIT/FT0/macros/FT0Misaligner.C:51]: I meant if you are not planning to align individual modules (radiatrors?) then there is no particular reason to define them as alignable. BTW, you can still have them in ideal position applying the AlignParam made of 0s.

Comment (AllaMaevskaya) [reply Detectors/FIT/FT0/macros/FT0Misaligner.C:51]: I will keep them for a while, OK?

Comment (shahor02) [reply Detectors/FIT/FT0/macros/FT0Misaligner.C:51]: ok

**Comment by shahor02** [comment]
@AllaMaevskaya could you resolve the conflict?

**Comment by AllaMaevskaya** [comment]
Errors are related to my code

**Comment by AllaMaevskaya** [comment]
I don't think that this relate to me code:
112/378 Test  #24: GPU/TPCFastTransformation/macro/fastTransformQA.C_compiled .....................***Failed   47.67 sec
=== GPU/TPCFastTransformation/macro/fastTransformQA.C_compiled - Starting test. Max attempts: 3. Timeout per attempt: 100. ===
=== GPU/TPCFastTransformation/macro/fastTransformQA.C_compiled - Argument: --name ===
=== GPU/TPCFastTransformation/macro/fastTransformQA.C_compiled - Argument: GPU/TPCFastTransformation/macro/fastTransformQA.C_compiled ===
--
  "_main", referenced from:
     implicit entry/start for main executable
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

Exit code=1
=== GPU/TPCFastTransformation/macro/fastTransformQA.C_compiled - Test attempt 1/3 failed with exit code 1 ===
--
  "_main", referenced from:
     implicit entry/start for main executable
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

**Pull request merged**